### DISCUSSION TITLE
Switch workspace & focus on application action

### DISCUSSION THREAD

**Discussion by Kibouo** [leftwm/leftwm#783] [Q&A]
Say I click on a link in workspace 1 while Firefox is opened on workspace 2. Is it possible to automatically switch to workspace 2 and focus on Firefox? Similarly, when Discord or Signal is opened in a workspace and one clicks their notification (using `dunst`), it makes those applications perform the correct action (e.g. switch channel). However, the workspace and focus is not switched.

My focus settings are:
```toml
focus_behaviour = "Sloppy"
focus_new_windows = true
sloppy_mouse_follows_focus = true
```

```bash
~/Downloads
 ; leftwm-check                                     
:: LeftWM version: 0.3.0
:: LeftWM git hash: eaf531d
:: Loading configuration . . .
    -> Configuration loaded OK 
:: Checking keybinds . . .
    -> All keybinds OK
:: Checking environment . . .
    -> Environment OK 
:: Checking theme . . .
ERROR: No theme folder or symlink `current` found.
```

**Comment by VuiMuich** [comment]
Welcome adventurer!

Sorry for this late reply, only about to catch up with things happening lately.

Afaik we have no means to catch such events at the moment. I guess for the url thing it might work if we could process the 'attention request' by the browser and actually the notification might link to its origin window in a similar way.

Pinging @AethanFoot as he is the most adapt around for anything x11 and EWMH relatead. (sorry Aethan :man_shrugging:)

**Comment by Kibouo** [comment]
Reacting to 'attention requests' is indeed a concise description of my question.

Disclaimer: I know nothing of EWMH. The Wikipedia page mentions the existence of an `Urgent` flag that seems to do exactly what I'm asking for though (maybe?). If this ever turns into a feature (request), such generic approach would likely be preferred.
